packages/clay-core/docs/treeview.mdx

```diff
@@ -0,0 +1,378 @@
+		new Set(['1', '2', '3'])
+	);
+
+	return (
+		<TreeView
+			expandedKeys={expandedKeys}
+			onExpandedChange={setExpandedKeys}
+		>
+			{...}
+		</TreeView>
+	);
+}
+```
+
+### Custom expander
+
+Customizing the expander is possible using the `expanderIcons` property, changing icons has an effect on the entire tree, it is not possible to change exclusively for a single Item.
+
+```jsx
+<TreeView
+	expanderIcons={{
+		close: <Icon symbol="hr" />,
+		open: <Icon symbol="plus" />,
+	}}
+>
+	{...}
+</TreeView>
+```
+
+## Multiple Selection
+
+The multi-selection in TreeView is a composition with the `<Checkbox />` component that must be added explicitly in the Item rendering.
+
+```jsx
+<TreeView.Item>
+	<Checkbox />
+	Drive
+</TreeView.Item>
+
+<TreeView.Item>
+	<TreeView.ItemStack>
+		<Checkbox />
+		Drive
+	</TreeView.ItemStack>
+	<TreeView.Group>
+		{...}
+	</TreeView.Group>
+</TreeView.Item>
+```
+
+It is not necessary to add the `onChange` event or the `checked` property, the TreeView adds the methods to the component under the covers, this allows you to just add the Checkbox in any order you want.
+
+Multi-selection is also controlled in the same way as Expander, it uses the `key` of the item to select the Item, the `selectedKeys` property and the `onSelectionChange` event are used for this purpose.
+
+```jsx
+function Example() {
+	const [selectedKeys, setSelectionChange] = useState(
+		new Set(['1', '2', '3'])
+	);
+
+	return (
+		<TreeView
+			onSelectionChange={setSelectionChange}
+			selectedKeys={selectedKeys}
+		>
+			{...}
+		</TreeView>
+	);
+}
+```
+
+### Recursive selection
+
+By default when selecting an Item with nested items, the items are recursively selected.
+
+There are some limitations, for static content the recursive selection only works if the item is expanded, ie visible. For dynamic content, it works independently.
+
+<div class="clay-site-alert alert alert-warning">
+	<strong class="lead">Warning</strong>
+	Recursive selection will not select items from an asynchronous Item.
+</div>
+
+<MultipleSelection />
+
+## Drag and Drop
+
+The Drag and Drop implementation handles items internally and works only for TreeView with dynamic content because it depends on the `items` property and must be enabled using the `dragAndDrop` property.
+
+The TreeView handles `items` immutably but follows a partial tree immutability implementation that is more optimized to do at runtime, to access these changes and be able to save in some service, you must control the state using the `items` property and the `onItemsChange` event.
+
+```jsx
+function Example() {
+	const [items, setItems] = useState([]);
+
+	return (
+		<TreeView
+			dragAndDrop
+			items={items}
+			onItemsChange={setItems}
+		>
+			{...}
+		</TreeView>
+	);
+}
+```
+
+## Shortcuts
+
+The TreeView implements shortcuts and manages the focus, and some shortcuts and focus trigger some shortcuts like renaming, removing, or asynchronous loading if any.
```

Comment:
@matuzalemsteles I don't really understand what we mean by
> and some shortcuts and focus trigger some shortcuts like renaming

I would change this to 

The TreeView implements shortcuts and manages the focus. Some shortcuts and focus trigger some actions like renaming, removing or asynchronous loading if any.

But I'd like to know what you think.

Reply:
Oh yeah, what I wrote got confused 😅, your suggestion got better to understanding haha

Reply:
Glad you like it